modules/tensor_mechanics/include/materials/ComputeInstantaneousThermalExpansionFunctionEigenstrain.h

```diff
@@ -26,14 +26,15 @@ class ComputeInstantaneousThermalExpansionFunctionEigenstrainTempl
 
 protected:
   virtual void initQpStatefulProperties() override;
-  virtual void computeThermalStrain(GenericReal<is_ad> & thermal_strain,
-                                    Real * dthermal_strain_dT) override;
+
+  virtual ValueAndDerivative<is_ad> computeThermalStrain() override;
 
   const Function & _thermal_expansion_function;
 
-  /// Stores the thermal strain as a scalar for use in computing an incremental update to this.
-  /// this is not a dual number as only the previous timestep values are used
-  //@{
+  /**
+   *@{ Stores the thermal strain as a scalar for use in computing an incremental update to this.
+   * this is not a dual number as only the previous timestep values are used
```

Comment:
After talking to @recuero we agree that there is no drawback in making this `GenericReal<is_ad>` (which would make this property potentially more useful and align with user's expectations and ensure interoperability with other objects that might calculate or consume a thermal strain property)